lib/ruby_units/unit.rb

```diff
@@ -31,30 +31,41 @@ class Unit < Numeric
     @@unit_match_regex = nil
     UNITY              = '<1>'.freeze
     UNITY_ARRAY        = [UNITY].freeze
-    # ideally we would like to generate this regex from the alias for a 'feet' and 'inches', but they aren't
-    # defined at the point in the code where we need this regex.
-    FEET_INCH_UNITS_REGEX = /(?:'|ft|feet)\s*(\d+)\s*(?:"|in|inch(?:es)?)/
-    FEET_INCH_REGEX    = /(\d+)\s*#{FEET_INCH_UNITS_REGEX}/
-    # ideally we would like to generate this regex from the alias for a 'pound' and 'ounce', but they aren't
-    # defined at the point in the code where we need this regex.
-    LBS_OZ_UNIT_REGEX  = /(?:#|lbs?|pounds?|pound-mass)+[\s,]*(\d+)\s*(?:ozs?|ounces?)/
-    LBS_OZ_REGEX       = /(\d+)\s*#{LBS_OZ_UNIT_REGEX}/
-    # ideally we would like to generate this regex from the alias for a 'stone' and 'pound', but they aren't
-    # defined at the point in the code where we need this regex.
-    # also note that the plural of 'stone' is still 'stone', but we accept 'stones' anyway.
-    STONE_LB_UNIT_REGEX = /(?:sts?|stones?)+[\s,]*(\d+)\s*(?:#|lbs?|pounds?|pound-mass)*/
-    STONE_LB_REGEX     = /(\d+)\s*#{STONE_LB_UNIT_REGEX}/
-    TIME_REGEX         = /(\d+)*:(\d+)*:*(\d+)*[:,]*(\d+)*/
-    SCI_NUMBER         = /([+-]?\d*[.]?\d+(?:[Ee][+-]?)?\d*)/
-    RATIONAL_NUMBER    = %r{\(?([+-])?(\d+[ -])?(\d+)\/(\d+)\)?}
-    COMPLEX_NUMBER     = /#{SCI_NUMBER}?#{SCI_NUMBER}i\b/
-    NUMBER_REGEX       = /#{SCI_NUMBER}*\s*(.+)?/
-    UNIT_STRING_REGEX  = %r{#{SCI_NUMBER}*\s*([^\/]*)\/*(.+)*}
-    TOP_REGEX          = /([^ \*]+)(?:\^|\*\*)([\d-]+)/
-    BOTTOM_REGEX       = /([^* ]+)(?:\^|\*\*)(\d+)/
-    NUMBER_UNIT_REGEX  = /#{SCI_NUMBER}?(.*)/
-    COMPLEX_REGEX      = /#{COMPLEX_NUMBER}\s?(.+)?/
-    RATIONAL_REGEX     = /#{RATIONAL_NUMBER}\s?(.+)?/
+    # ideally we would like to generate this regex from the alias for a 'feet'
+    # and 'inches', but they aren't defined at the point in the code where we
+    # need this regex.
+    FEET_INCH_UNITS_REGEX = /(?:'|ft|feet)\s*(\d+)\s*(?:"|in|inch(?:es)?)/.freeze
```

Comment:
Mostly freezing regexes and adding documentation. This way we don't redefine a bunch of Regexes that get used a lot every time we parse a unit.